Detectors/GlobalTracking/src/MatchHMP.cxx

```diff
@@ -377,41 +398,39 @@ void MatchHMP::doMatching()
       auto& trefTrk = trackWork.first;
 
       prop->getFieldXYZ(trefTrk.getXYZGlo(), bxyz);
-      Double_t bz = -bxyz[2];
+      double bz = -bxyz[2];
 
-      double timeUncert = 100.; //= trackWork.second.getTimeStampError();
+      double timeUncert = trackWork.second.getTimeStampError();
 
-      float minTrkTime = (trackWork.second.getTimeStamp() - mSigmaTimeCut * timeUncert) * 1.E3; // minimum track time in ns
-      float maxTrkTime = (trackWork.second.getTimeStamp() + mSigmaTimeCut * timeUncert) * 1.E3; // maximum track time in ns
+      float minTrkTime = (trackWork.second.getTimeStamp() - mSigmaTimeCut * timeUncert);
+      float maxTrkTime = (trackWork.second.getTimeStamp() + mSigmaTimeCut * timeUncert);
 
-      if (evtTime < (maxTrkTime + timeFromTF) && evtTime > (minTrkTime + timeFromTF)) {
+      // if (evtTime < (maxTrkTime + timeFromTF) && evtTime > (minTrkTime + timeFromTF)) {
+      if (evtTime < maxTrkTime && evtTime > minTrkTime) {
 
         evtTracks++;
 
-        MatchInfo* matching = new o2::dataformats::MatchInfoHMP(999999, mTrackGid[type][cacheTrk[itrk]]);
+        MatchInfo matching = MatchInfo(999999, mTrackGid[type][cacheTrk[itrk]]);
```

Comment:
use `MatchInfo matching(999999, mTrackGid[type][cacheTrk[itrk]]);` to avoid extra assignment.